sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -99,17 +97,17 @@ object TableOutputResolver {
         }
         (matchedCol.dataType, expectedCol.dataType) match {
           case (matchedType: StructType, expectedType: StructType) =>
-            checkNullability(matchedCol, expectedCol, conf, addError, newColPath)
```

Comment:
Moved nullability checks inside `resolveXXX` methods to share them with the position-based path.